src/AdoNet/Shared/Storage/DbConnectionFactory.cs

```diff
@@ -40,21 +51,25 @@ private static CachedFactory GetFactory(string invariantName)
                 throw new ArgumentNullException(nameof(invariantName));
             }
 
-            Tuple<string, string> providerFactoryDefinition;
+            Tuple<List<string>, string> providerFactoryDefinition;
             if (!providerFactoryTypeMap.TryGetValue(invariantName, out providerFactoryDefinition))
                 throw new InvalidOperationException($"Database provider factory with '{invariantName}' invariant name not supported.");
 
-            Assembly asm;
-            try
+            Assembly asm = null;
+            for (int asmIndex = 0; asmIndex < providerFactoryDefinition.Item1.Count; ++asmIndex)
             {
-                var asmName = new AssemblyName(providerFactoryDefinition.Item1);
-                asm = Assembly.Load(asmName);
+                try
+                {
+
+                    var asmName = new AssemblyName(providerFactoryDefinition.Item1[asmIndex]);
+                    asm = Assembly.Load(asmName);
+                }
+                catch (Exception exc)
+                {
+                    Console.WriteLine($"Unable to find and/or load a candidate assembly for '{invariantName}' invariant name.", exc);
```

Comment:
Should be taken off, suggestions what could be done and what would be the future proof mechanism for Orleans logging if not possible now?